pengine/allocate.c

```diff
@@ -1421,11 +1422,15 @@ stage6(pe_working_set_t * data_set)
                 dc_down = stonith_op;
                 dc_fence = stonith_op;
 
-            } else {
+            } else if (is_set(data_set->flags, pe_flag_concurrent_fencing) == FALSE) {
                 if (last_stonith) {
                     order_actions(last_stonith, stonith_op, pe_order_optional);
                 }
                 last_stonith = stonith_op;
+
+            } else {
+                order_actions(stonith_op, done, pe_order_implies_then);
+                stonith_ops = g_list_append(stonith_ops, stonith_op);
```

Comment:
I think I kind of wanted to avoid changing the default behavior, which would introduce the updates of a lot of regression tests. But sure, if we _really_ want to make it the default and simplify the configuration.

The resource parameter pcmk_action_limit specifies the per-device limit. We'd probably still need the global concurrent-fencing even if no device has pcmk_action_limit > 1.